pyqtorch/time_dependent/solvers.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import warnings
+from typing import Any, Callable
+
+import torch
+from torch import Tensor
+
+from pyqtorch.time_dependent.integrators.adaptive import AdaptiveIntegrator
+from pyqtorch.time_dependent.methods.dp5 import DormandPrince5
+from pyqtorch.time_dependent.methods.krylov import Krylov
+from pyqtorch.time_dependent.options import AdaptiveSolverOptions
+from pyqtorch.utils import cache
+
+
+class SESolver(AdaptiveIntegrator):
+    """Generic adaptive-step solver for Schrodinger equation."""
+
+    def __init__(
+        self,
+        H: Tensor | Callable[..., Any],
+        psi0: Tensor,
+        tsave: Tensor,
+        options: AdaptiveSolverOptions,
+    ):
+        super().__init__(H, psi0, tsave, options)
+
+    def odefun(self, t: float, psi: Tensor) -> Tensor:
+        """Compute dpsi / dt = -1j * H(psi) at time t."""
+        with warnings.catch_warnings():
+            # filter-out UserWarning about "Sparse CSR tensor support is in beta state"
+            warnings.filterwarnings("ignore", category=UserWarning)
```

Comment:
I think we should use the new logging infra @dominikandreasseitz ?

Reply:
you mean to log this warning?